pkg/apiserver/registry/inventory/rest.go

```diff
@@ -0,0 +1,185 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/client-go/tools/cache"
+
+	runtimev1alpha1 "antrea.io/nephe/apis/runtime/v1alpha1"
+	"antrea.io/nephe/pkg/controllers/inventory"
+)
+
+// REST implements rest.Storage for Vpc.
+type REST struct {
+	vpcIndexer cache.Indexer
+	logger     logger.Logger
+}
+
+var (
+	_ rest.Scoper = &REST{}
+	_ rest.Getter = &REST{}
+	_ rest.Lister = &REST{}
+)
+
+// NewREST returns a REST object that will work against API services.
+func NewREST(indexer cache.Indexer, l logger.Logger) *REST {
+	return &REST{
+		vpcIndexer: indexer,
+		logger:     l,
+	}
+}
+
+func (r *REST) New() runtime.Object {
+	return &runtimev1alpha1.Vpc{}
+}
+
+func (r *REST) NewList() runtime.Object {
+	return &runtimev1alpha1.VpcList{}
+}
+
+func (r *REST) Get(ctx context.Context, name string, _ *metav1.GetOptions) (runtime.Object, error) {
+	ns, ok := request.NamespaceFrom(ctx)
+	if !ok || len(ns) == 0 {
+		return nil, errors.NewBadRequest("Namespace cannot be empty.")
+	}
+	namespacedName := ns + "/" + name
+
+	objs, err := r.vpcIndexer.ByIndex(inventory.VpcIndexerByVpcNamespacedName, namespacedName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(objs) == 0 {
+		return nil, errors.NewNotFound(runtimev1alpha1.Resource("vpc"), name)
+	}
+	vpc := objs[0].(*runtimev1alpha1.Vpc)
+	return vpc, nil
+}
+
+func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	// List only supports four types of input options
+	// 1. All namespace
+	// 2. Labelselector with only the specific namespace, the only valid labelselectors are "account-name=<accountname>" and "region=<region>"
+	// 3. Fieldselector with only the specific namespace, the only valid fieldselectors is "metadata.name=<metadata.name>"
+	// 4. Specific Namespace
+	accountName := ""
+	region := ""
+	if options != nil && options.LabelSelector != nil && options.LabelSelector.String() != "" {
+		labelSelectorStrings := strings.Split(options.LabelSelector.String(), ",")
+		for _, labelSelectorString := range labelSelectorStrings {
+			labelKeyAndValue := strings.Split(labelSelectorString, "=")
+			if labelKeyAndValue[0] == "account-name" {
+				accountName = labelKeyAndValue[1]
+			} else if labelKeyAndValue[0] == "region" {
+				region = labelKeyAndValue[1]
+			} else {
+				return nil, errors.NewBadRequest("unsupported label selector, supported labels are: account-name and region")
+			}
+		}
+	}
+
+	name := ""
+	if options != nil && options.FieldSelector != nil && options.FieldSelector.String() != "" {
+		fieldSelectorStrings := strings.Split(options.FieldSelector.String(), ",")
+		for _, fieldSelectorString := range fieldSelectorStrings {
+			fieldKeyAndValue := strings.Split(fieldSelectorString, "=")
+			if fieldKeyAndValue[0] == "metadata.name" {
```

Comment:
add field selector for namespace too

Reply:
@reachjainrahul when the Users use field selector, we force them to enter "-n <namespace>", they have already entered namespace, do we still need field selector for namespace?

Reply:
I see typically usage as name and namespace.. lets not force them to add -n.. If they want namespace in field selector, we should allow that

Reply:
@reachjainrahul  As discussed before, add "-n <namespace>" is necessary for the List(), because all of us search by Namespace+selector(name, accountname, region...), if the namespace is all-namespace, then we cannot use the Byindex anymore because we don't have the specific namespace. 

But maybe if we want "metadata.namespace" still can be added if want,but what if the "metadata.namespace"  and the "-n <namespace>" is different?